mock_hardware/src/generic_system.cpp

```diff
@@ -237,6 +243,83 @@ CallbackReturn GenericSystem::on_init(const hardware_interface::HardwareInfo & i
     initialize_storage_vectors(gpio_commands_, gpio_states_, gpio_interfaces_, info_.gpios);
   }
 
+  actuator_slowdown_ = std::stod(info_.hardware_parameters["actuator_slowdown"]);
+
+  const auto num_joints = std::accumulate(
+    info_.transmissions.begin(), info_.transmissions.end(), 0ul,
+    [](const auto & acc, const auto & trans_info) { return acc + trans_info.joints.size(); });
+
+  const auto num_actuators = std::accumulate(
+    info_.transmissions.begin(), info_.transmissions.end(), 0ul,
+    [](const auto & acc, const auto & trans_info) { return acc + trans_info.actuators.size(); });
```

Comment:
OK, I think I understand why do we need both. Since there i m:n relationship between transmission and joints. So, OK, having two lists should be fine.